cg/meta/invoice.py

```diff
@@ -108,17 +135,25 @@ def get_price(self, discount: int, record: models.Sample):
         if not discount:
             discount=0
         discount_factor = float(100 - discount) / 100
-
+        if not full_price:
+            return None
         return full_price * discount_factor
 
 
-    def total_price(self, invoice_obj: models.Invoice) -> float:
-        discount = invoice_obj.discount
+    def total_price(self) -> float:
+        discount = self.invoice_obj.discount
         total_price = 0
-        if invoice_obj.pools:
-            for record in invoice_obj.pools:
-                total_price += self.get_price(discount, record)
-        elif invoice_obj.samples:
-            for record in invoice_obj.samples:
-                total_price += self.get_price(discount, record)
+        if self.invoice_obj.pools:
+            for record in self.invoice_obj.pools:
+                if self.get_price(discount, record):
+                    total_price += self.get_price(discount, record)
+                else:
+                    return ''
```

Comment:
Since python 3, we are able to give hints as to what the return type of a function is. What is the return type of this function? :)